server/logConverter.js

```diff
@@ -0,0 +1,136 @@
+const spawn = require("child_process").spawn;
+const path = require('path')
+const appRoot = require('app-root-path')
+const fetch = require('node-fetch');
+
+class logConverter {
+  constructor (settings, winston) {
+    this.options = {
+      // the interval of sync, every 2h sec
+      interval: 20
+    }
+    
+    this.winston = winston
+
+    this.pythonFolder = path.join(appRoot.toString(), 'python')
+    this.pythonScript = path.join(this.pythonFolder, 'tlog2kmz.py')
+
+    this.converterPid = null
+    this.tlogfilename = null
+
+    process.on('SIGINT', this.quitting) // run signal handler on CTRL-C
+    process.on('SIGTERM', this.quitting) // run signal handler on SIGTERM
+    process.on('exit', this.quitting) // run signal handler when main process exits
+
+    // load settings
+    this.settings = settings
+    this.options.doLogConversion = this.settings.value('logConverter.doLogConversion', true)
+
+    // interval for conversion checks
+    this.intervalObj = setInterval(() => {
+      console.log('LogConverter interval')
+      if (this.options.doLogConversion) {
+      
+        try {
+          this.getTlogFilename()
+          // console.log(this.tlogfilename)
+          var requestActiveTlog =  JSON.stringify(this.tlogfilename)
+        } catch (error) {
+          console.error(error);
+        } finally {
+          try{
+              console.log('Doing log conversion...')
+              if (requestActiveTlog){
+                var fileToIgnore = JSON.parse(requestActiveTlog)
+                if (fileToIgnore){
+                  if (fileToIgnore.tlogfilename != null ){
+                    // Ignore tlog in use
+                    this.converterPid = spawn('python3',[this.pythonScript, '--ignore', fileToIgnore.tlogfilename]);     
```

Comment:
So my understanding here is that you don't want to convert the log-in-progress. I suspect it'd be easier to do this in Python rather than nodeJS. Something like "if logfile last modified time is less than 20 sec ago, don't convert"

Reply:
Moved to Python. I hope cleared the JS about that.